From 98a3b25143161e249102eceef6face6833ba19ea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Mon, 12 Oct 2020 19:11:44 +0200
Subject: [PATCH 07/28] widl: Don't output typedef statement in WinRT mode.

For types under a non-global namespace, since we reference them with
their prefixed mangled names.
---
 tools/widl/header.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tools/widl/header.c b/tools/widl/header.c
index d8dfcf4f4c3..0a8272fdbb4 100644
--- a/tools/widl/header.c
+++ b/tools/widl/header.c
@@ -813,6 +813,8 @@ static void write_generic_handle_routines(FILE *header)
 
 static void write_typedef(FILE *header, type_t *type, int declonly)
 {
+  type_t *t = type_alias_get_aliasee_type(type);
+  if (winrt_mode && t->namespace && !is_global_namespace(t->namespace)) return;
   fprintf(header, "typedef ");
   write_type_v(header, type_alias_get_aliasee(type), FALSE, declonly, type->name, NAME_C);
   fprintf(header, ";\n");
-- 
2.28.0

